pac4j-core/src/main/java/org/pac4j/core/profile/ProfileManager.java

```diff
@@ -54,15 +54,23 @@ public ProfileManager(final WebContext context) {
      * @return the map of profiles
      */
     private LinkedHashMap<String, U> retrieveAll(final boolean readFromSession) {
-        LinkedHashMap<String, U> profiles = new LinkedHashMap<>();
-        final Object objSession = this.context.getRequestAttribute(Pac4jConstants.USER_PROFILES);
-        if (objSession != null && objSession instanceof LinkedHashMap) {
-            profiles = (LinkedHashMap<String, U>) objSession;
+        final LinkedHashMap<String, U> profiles = new LinkedHashMap<>();
+        final Object request = this.context.getRequestAttribute(Pac4jConstants.USER_PROFILES);
+        if (request != null) {
+            if  (request instanceof LinkedHashMap) {
+                profiles.putAll((LinkedHashMap<String, U>) request);
+            }
+            if (request instanceof CommonProfile) {
+                profiles.put(((CommonProfile) request).getClientName(), (U) request);
```

Comment:
I am confused. This is targeted to 1.9.1 right? Why would the call ever return `null`?